### PR TITLE
interface for ESP wifi module added

### LINKED ISSUE #1891: Add feature to connect to PSLab over wifi using ESP module
**Actual Behaviour**
Currently no feature to connect to PSLab over wifi 

**Expected Behaviour**
Add feature to connect to PSLab over wifi using ESP

**Steps to reproduce it**
N/A

**LogCat for the issue**
N/A

**Screenshots of the issue**
N/A

**Would you like to work on the issue?**
Yes

### PR DESCRIPTION
**Pull request by neel1998**
Fixes #1891 

**Changes**: User can connect to pslab using wifi

**Screenshot/s for the changes**:

**Checklist**: [Please tick following check boxes with `[x]` if the respective task is completed]
- [x] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [x] No modifications done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [x] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [x] My code does not contain any extra lines or extra spaces
- [x] I have requested reviews from other members

**APK for testing**: 
[esp.zip](https://github.com/fossasia/pslab-android/files/3472336/esp.zip)

@CloudyPadmal  the interface seems nice, Once the ESP server is implemented we can add further communication protocols.

### CODE REVIEW

**Comment by CloudyPadmal** [comment]
Third time I'm seeing you closing and opening new PRs. 

This PR is related to #1892

File: app/build.gradle [hunk 1]
```diff
@@ -139,4 +139,5 @@ dependencies {
         exclude group: 'com.android.support', module: 'appcompat-v7'
         exclude group: 'com.android.support', module: 'design'
     }
+    implementation "com.squareup.okhttp3:okhttp:4.0.1"
```
Comment (CloudyPadmal) [inline app/build.gradle]: Move where the other dependencies are and set version in config files.

**Comment by neel1998** [comment]
I am sorry but while trying out the BFG tool ... I had accidentally removed png and XML files from many of my local branches.. if made a new commit using the previous branch....All the png and XML would be again added to the commit. That's why I had to open another PR from new branch. Sorry for the inconvenience.

File: app/src/main/java/io/pslab/activity/MainActivity.java [hunk 2]
```diff
@@ -406,22 +408,51 @@ private void attemptToConnectPSLab() {
...
                         .setPositiveButton(getResources().getString(R.string.bluetooth_scan_text), new DialogInterface.OnClickListener() {
                             @Override
                             public void onClick(DialogInterface dialog, int which) {
-                                BluetoothScanFragment bluetoothScanFragment = new BluetoothScanFragment();
-                                bluetoothScanFragment.show(getSupportFragmentManager(), "bluetooth");
-                                bluetoothScanFragment.setCancelable(true);
+                                AlertDialog.Builder dialogBuilder = new AlertDialog.Builder(MainActivity.this);
+                                LayoutInflater inflater = MainActivity.this.getLayoutInflater();
+                                View dialogView = inflater.inflate(R.layout.bluetooth_wifi_dialog_layout, null);
+                                dialogBuilder.setNegativeButton("Cancel", new DialogInterface.OnClickListener() {
```
Comment (CloudyPadmal) [inline app/src/main/java/io/pslab/activity/MainActivity.java]: Hardcoded

File: app/src/main/java/io/pslab/fragment/ESPFragment.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        Button espConnectBtn = rootView.findViewById(R.id.esp_connect_btn);
+        espConnectProgressBar = rootView.findViewById(R.id.esp_connect_progressbar);
+        espConnectBtn.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                espIPAddress = espIPEditText.getText().toString();
+                if (espIPAddress.length() == 0) {
+                    Toast.makeText(getContext(), getResources().getString(R.string.incorrect_IP_address_message), Toast.LENGTH_SHORT).show();
+                } else {
+                    espConnectProgressBar.setVisibility(View.VISIBLE);
```
Comment (CloudyPadmal) [inline app/src/main/java/io/pslab/fragment/ESPFragment.java]: Set in `onPreExecute`

File: app/src/main/java/io/pslab/fragment/ESPFragment.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        ViewGroup.LayoutParams params = getDialog().getWindow().getAttributes();
+        params.height = ViewGroup.LayoutParams.WRAP_CONTENT;
+        params.width = ViewGroup.LayoutParams.MATCH_PARENT;
+        getDialog().getWindow().setAttributes((android.view.WindowManager.LayoutParams) params);
+    }
+    private class ESPTask extends AsyncTask<Void, Void, String> {
```
Comment (CloudyPadmal) [inline app/src/main/java/io/pslab/fragment/ESPFragment.java]: Reformat the whole code to add proper indents and spaces. Should be quiet obvious by now.

File: app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -18,12 +18,16 @@
     <string name="pslab_pinlayout_front">Pin Layout Front</string>
     <string name="pslab_pinlayout_back">Pin Layout Back</string>
     <string name="bluetooth_connect_menu">Bluetooth Connection</string>
-    <string name="bluetooth_scan_dialog">USB device not found. Do you want to scan for Bluetooth Connection?</string>
-
+    <string name="bluetooth_wifi_scan_dialog">USB device not found. Do you want to scan for Bluetooth or Wifi Connection?</string>
```
Comment (CloudyPadmal) [inline app/src/main/res/values/strings.xml]: Connection -> connection

File: app/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -18,12 +18,16 @@
...
+    <string name="bluetooth">Bluetooh</string>
```
Comment (CloudyPadmal) [inline app/src/main/res/values/strings.xml]: Bluetooth

**Comment by CloudyPadmal** [comment]
> I am sorry but while trying out the BFG tool ... I had accidentally removed png and XML files from many of my local branches.. if made a new commit using the previous branch....All the png and XML would be again added to the commit. That's why I had to open another PR from new branch. Sorry for the inconvenience.

Thing is, when you're making new PRs closing older ones, whole conversations are gone. All the discussions, changes requested; just gone.

Whatever the issue you're having with your local branch, fix it before making any more PRs.

**Comment by neel1998** [comment]
> > I am sorry but while trying out the BFG tool ... I had accidentally removed png and XML files from many of my local branches.. if made a new commit using the previous branch....All the png and XML would be again added to the commit. That's why I had to open another PR from new branch. Sorry for the inconvenience.
> 
> Thing is, when you're making new PRs closing older ones, whole conversations are gone. All the discussions, changes requested; just gone.
> 
> Whatever the issue you're having with your local branch, fix it before making any more PRs.

I totally understand your point. And I was using the tool for the first time so made such mistake. Again really sorry for this, won't happen again.

**Pull request merged**